### PR TITLE
industry/foc: add support for Hall sensor and Qencoder feedback

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary
The main goal of this PR is to add Hall sensor and Qencoder feedback for FOC. 

## Impact
New functionality.

## Testing
Tested with modified example/foc app.

### CODE REVIEW

File: include/industry/foc/foc_common.h [hunk 1]
```diff
@@ -71,6 +71,15 @@ enum foc_ramp_mode_e
   RAMP_MODE_NORMAL    = 3,      /* Normal operation */
 };
 
+/* Angle handler type */
+
+enum foc_angle_type_e
+{
+  FOC_ANGLE_TYPE_INVALID = 0,  /* Reserved */
+  FOC_ANGLE_TYPE_ELE     = 1,  /* Electrical angle */
+  FOC_ANGLE_TYPE_MECH    = 2,  /* Mechanical angle */
```
Comment (acassis) [inline include/industry/foc/foc_common.h:80]: @raiden00pl why do we need to have two separated types of FOC Angles ?

Comment (raiden00pl) [reply include/industry/foc/foc_common.h:80]: There are 2 possible motor angle sensors:
1. the electrical angle sensors (which is the angle between motor poles) like Hall-sensor or observer-based angle estimation
2. the mechanical angle sensors (which is the angle of motor shaft vs motor stator) like encoder  
The FOC algorithm requires an electrical angle as input, so we must known if the value obtained requires conversion to an electrical angle or not.

Comment (acassis) [reply include/industry/foc/foc_common.h:80]: Got it! Thank you!

Comment (uLipe) [reply include/industry/foc/foc_common.h:80]: Just adding some extra information, having the number of poles which is tipically given the foc controller is able to convert the mechanical angle into the electrical one.

The only extra requirement that may have to be considered is a initial park and go alignment to keep the IQ + ID pair aligned to a initial known value and consider this offset to the account when converting mechanical into electrical angle.

**Pull request merged**